apollo-common/src/main/java/com/ctrip/framework/apollo/common/utils/ConfigValidator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.common.utils;
+
+import com.ctrip.framework.apollo.core.utils.StringUtils;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.util.Assert;
+
+/**
+ * Config Validator
+ *
+ * @author Lv Lifeng
+ */
+public class ConfigValidator {
+
+  private final ConfigurableApplicationContext applicationContext;
+
+  public ConfigValidator(ConfigurableApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+    checkDataBaseName();
+  }
+
+  private void checkDataBaseName() {
+    String dataSourceUrl = System.getProperty("spring.datasource.url");
+    if (!StringUtils.isBlank(dataSourceUrl)
+        && dataSourceUrl.contains("/")
+        && dataSourceUrl .contains("?")
+        && !StringUtils.isBlank(applicationContext.getId())) {
+      String dbName = dataSourceUrl.substring(dataSourceUrl.lastIndexOf("/"), dataSourceUrl.indexOf("?"));
+      if (!StringUtils.isEmpty(dbName)) {
+        switch (applicationContext.getId()) {
+          case "apollo-adminservice":
+          case "apollo-configservice":
+            Assert.isTrue(dbName.toLowerCase().contains("config"), "Please configure the" +
```

Comment:
Shall we add a flag to allow users to disable this check? I'm not sure whether all the current apollo installation follows the name pattern.

Reply:
Okay, I quite agree!

Reply:
I have added this flag,`config.check.databasename = false`,If this property is not configured or true, the check logic will be executed.
Since the database name can be changed by the user himself, would it be more friendly to give a prompt if the database name is configured incorrectly? Instead of failing to startup. for example, printing a warning log.